components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/apikey/ApiKeyAuthenticator.java

```diff
@@ -358,6 +361,71 @@ public AuthenticationResponse authenticate(MessageContext synCtx) {
         }
     }
 
+    private void validateAPIKeyRestrictions(JWTClaimsSet payload, MessageContext synCtx) throws APISecurityException {
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) synCtx).
+                getAxis2MessageContext();
+
+        String permittedIPList = (String) payload.getClaim(APIConstants.JwtTokenConstants.PERMITTED_IP);
+
+        if (StringUtils.isNotEmpty(permittedIPList)) {
```

Comment:
!= null and not empty

Reply:
StringUtils.isNotEmpty() method already checks for null. So, why do we need to check again?